### ISSUE TITLE
Memory usage

### ISSUE THREAD

**Issue by artplan1** [floere/phony#415]
First of all thanks for great gem.

In readme it [says](https://github.com/floere/phony#runtime-memory-usage):
> According to memory_profiler, the Phony gem uses roughly 1 MB of memory per Ruby process. Usage was generated using: ruby -e 'require "memory_profiler"; MemoryProfiler.report(allow_files: "phony"){ require "phony" }.pretty_print'

But as I see it uses ~2mb now, mostly because of `Germany` and `Russia_kazakhstan_abkhasia_south_ossetia` codes

This is what I see:

```
Total allocated: 2059039 bytes (24279 objects)
Total retained:  955137 bytes (16008 objects)

allocated memory by gem
-----------------------------------
   2059039  phony/lib

allocated memory by file
-----------------------------------
    594944  /Users/artem/projects/phony/lib/phony/countries/germany.rb
    242330  /Users/artem/projects/phony/lib/phony/national_splitters/variable.rb
    204280  /Users/artem/projects/phony/lib/phony/countries/russia_kazakhstan_abkhasia_south_ossetia.rb
    193472  /Users/artem/projects/phony/lib/phony/dsl.rb
    145874  /Users/artem/projects/phony/lib/phony/national_splitters/dsl.rb
    105016  /Users/artem/projects/phony/lib/phony/countries/united_kingdom.rb
     87904  /Users/artem/projects/phony/lib/phony/countries.rb
     72547  /Users/artem/projects/phony/lib/phony/trunk_code.rb
     41968  /Users/artem/projects/phony/lib/phony/countries/ukraine.rb
     40672  /Users/artem/projects/phony/lib/phony/country.rb
     37298  /Users/artem/projects/phony/lib/phony/countries/bangladesh.rb
....
```

of course 2MB is not huge load, but maybe it worths to update readme

Also in our project we're targeting US and CA as for now, we want to use phony's validators and normalizers but it also will be great to have ability to **configure** loaded countries. Let me know if it makes sense, I can provide PRs.

Thanks.

**Comment by floere** [comment]
> Also in our project we're targeting US and CA as for now, we want to use phony's validators and normalizers but it also will be great to have ability to configure loaded countries.

I'm not sure what you mean – you would like to change existing definitions? Have you tries simply doing the same as in `countries.rb` (`Phony.define do` etc.)?

**Comment by artplan1** [comment]
> > Also in our project we're targeting US and CA as for now, we want to use phony's validators and normalizers but it also will be great to have ability to configure loaded countries.
> 
> I'm not sure what you mean – you would like to change existing definitions? Have you tries simply doing the same as in `countries.rb` (`Phony.define do` etc.)?

Thank you for your responses.

I don't want to change definitions, but I want to exclude some countries from loading (or only include particular). In the meantime we need only US and CA (and we still want to use this gem), but right now we're loading everything else too.

**Comment by floere** [comment]
Ah I understand! And currently, when you `require "phony"`, you get everything else as well.

What you'd like is probably:

```
require "phony/core"
require "phony/countries"
```

or maybe

```
require "phony/core"
Phony.load_countries('1')
```

In any case you'd like to load exactly the country you'll be using – is that correct?

**Comment by artplan1** [comment]
that's correct. Some flexible list of required countries.

but anyways, it's just 2MB 👍(~40% of it is Germany and Russia_kazakhstan_abkhasia_south_ossetia 😀)

**Comment by nialbima** [comment]
Also interested in configuring the loaded countries. Is there a good way to do this in the current version?

**Comment by floere** [comment]
Sadly, not yet without monkey patching the code. I'm thinking of adding a:

```
require 'phony/config'

# Explicit
Phony::Config.load(only: ['41', '44'])
Phony::Config.load(except: ['41', '44'])
# Convenience
Phony::Config.load('41', '44')
# Or even
Phony::Config.load(41, 44)

# These then load Phony.
```

or similar. As @artplan1 said, since it's not huge, perhaps not huge/urgent issue, but I'd like to include it.

**Comment by floere** [comment]
@artplan1 A late P.S: retained memory usage was still under 1MB, as you can see with `Total retained:  955137 bytes (16008 objects)`.

But in any case, I've released 2.18.0, which allows for the syntax described above.

Load all (Same as `require 'phony'`):

```
ruby -e 'require "memory_profiler"; MemoryProfiler.report(allow_files: "phony"){ require "phony/config"; Phony::Config.load }.pretty_print'
Total allocated: 4513932 bytes (60534 objects)
Total retained:  1227879 bytes (17341 objects)
```

Just the NANP CC:

```
ruby -e 'require "memory_profiler"; MemoryProfiler.report(allow_files: "phony"){ require "phony/config"; Phony::Config.load("1") }.pretty_print'
Total allocated: 4488012 bytes (60531 objects)
Total retained:  63403 bytes (694 objects)
```

All except GER, RUS:

```
ruby -e 'require "memory_profiler"; MemoryProfiler.report(allow_files: "phony"){ require "phony/config"; Phony::Config.load(except: ["7", "45"]) }.pretty_print'
Total allocated: 4513740 bytes (60532 objects)
Total retained:  1116624 bytes (15094 objects)
```
